src/AzureIoTHub.Portal.Tests.Unit/Client/Pages/EdgeDevices/EdgeDeviceListPageTests.cs

```diff
@@ -313,9 +334,53 @@ public async Task FliterBySelectModelShould()
 
             var newModelList = await cut.Instance.Search("01");
 
+            await cut.Instance.ChangeModel(modelList[0]);
+
+            cut.WaitForElement("#searchFilterButton").Click();
+
             // Assert
             cut.WaitForAssertion(() => cut.Markup.Should().NotContain("Loading..."));
             Assert.AreEqual(1, newModelList.Count());
+            _ = cut.FindAll("table tbody tr").Count.Should().Be(1);
+            cut.WaitForAssertion(() => MockRepository.VerifyAll());
+        }
+
+        [Test]
+        public void ClickOnDeleteShouldShowDeleteDialog()
+        {
+            // Arrange
+            var expectedUrl = "api/edge/devices?pageNumber=0&pageSize=10&searchText=&searchStatus=&orderBy=&modelId=";
+            _ = this.mockEdgeDeviceClientService.Setup(service => service.GetDevices(expectedUrl))
+                .ReturnsAsync(new PaginationResult<IoTEdgeListItem>
+                {
+                    Items = new List<IoTEdgeListItem>
+                    {
+                        Fixture.Create<IoTEdgeListItem>()
+                    }
+                });
+
+            _ = this.mockEdgeModelClientService.Setup(service => service.GetIoTEdgeModelList())
+                .ReturnsAsync(new List<IoTEdgeModelListItem>()
+                {
+                    new IoTEdgeModelListItem()
+                    {
+                        Name = Guid.NewGuid().ToString(),
+                        Description = Guid.NewGuid().ToString(),
+                    }
+                });
+
+            var mockDialogReference = new DialogReference(Guid.NewGuid(), this.mockDialogService.Object);
+
+            _ = this.mockDialogService.Setup(c => c.Show<EdgeDeviceDeleteConfirmationDialog>(It.IsAny<string>(), It.IsAny<DialogParameters>()))
+                .Returns(mockDialogReference);
+
+            // Act
+            var cut = RenderComponent<EdgeDeviceListPage>();
+
+            var deleteIcon = cut.WaitForElement(".deleteDeviceIcon");
```

Comment:
Instead of using a Css class selector, please prefer an identifier on the button (not the Icon)